zerver/lib/topic.py

```diff
@@ -115,17 +115,19 @@ def user_message_exists_for_topic(user_profile: UserProfile,
     ).exists()
 
 def update_messages_for_topic_edit(message: Message,
+                                   client_oldest_message: Message,
                                    propagate_mode: str,
                                    orig_topic_name: str,
                                    topic_name: str) -> List[Message]:
     propagate_query = Q(recipient = message.recipient, subject = orig_topic_name)
-    # We only change messages up to 2 days in the past, to avoid hammering our
-    # DB by changing an unbounded amount of messages
+    # We only change messages up to 7 days in the past or all the messages
+    # which are sent after the oldest message visible to the client.
     if propagate_mode == 'change_all':
-        before_bound = timezone_now() - datetime.timedelta(days=2)
+        before_bound = timezone_now() - datetime.timedelta(days=7)
```

Comment:
Can you put the 2 => 7 change in its own commit?  It's really an independent change we're doing here.